src/daft-writers/src/batch.rs

```diff
@@ -0,0 +1,84 @@
+use std::sync::Arc;
+
+use common_buffer::{Bufferable, RowBasedBuffer};
+use common_error::DaftResult;
+use daft_table::Table;
+
+use crate::{FileWriter, WriterFactory};
+
+// TargetBatchWriter is a writer that writes in batches of rows, i.e. for Parquet where we want to write
+// a row group at a time. It uses a buffer to accumulate rows until it has enough to write a batch.
+pub struct TargetBatchWriter<B: Bufferable> {
```

Comment:
Todo: write unit tests for these, we can make `DummyTestWriters` that do nothing